### DISCUSSION TITLE
Support more audio codecs, not just lossy MP3 - in ESPHome

### DISCUSSION THREAD

**Discussion by nagyrobi** [music-assistant/support#889] [Feature requests and ideas]
ESPHome's Media Player currently seems to support only MP3 streams. MP3 is a lossy format, sound quality is degraded.
Music Assistant integration by default renders the audio streams in FLAC, which is a lossless format.
Would be nice to enable support in ESPHome for other codecs too:
- lossless FLAC
- uncompressed PCM up to 48k stereo
- open source OPUS (which is lossy, but way better than MP3)
- AAC (also lossy, but better than MP3 and OPUS)

Since only MP3 is supported, one needs to use other firmware than ESPHome, like squeezelite. The goal would be not to have to rely on third party stuff, but remain in the Home Assistant + ESPHome ecosystem.

Adding more codecs of course needs PSRAM for ESP32 (like WROVER module), which is not a problem as plenty of such modules are available. This just needs to be pointed out properly in the documentation.

Also requested in ESPHome FR repo: https://github.com/esphome/feature-requests/issues/1848

**Comment by nagyrobi** [comment]
https://github.com/esphome/feature-requests/issues/1875

**Comment by Hedda** [comment]
FYI, Nabu Casa developers have as part of firmware for the Home Assistant Voice Preview Edition developed a brand new ”nabu” media player custom component which has most of those features and that code will hopefully be pushed upstream to mainline ESPHome soon, see:

https://github.com/esphome/home-assistant-voice-pe/tree/dev/esphome/components/nabu

And

https://www.home-assistant.io/blog/2024/12/19/voice-preview-edition-the-era-of-open-voice/

They already moved some of the new audio decoder and resampling libraries from that repository into their own separate repo that can be found here:

https://github.com/esphome/esp-audio-libs

**Comment by OzGav** [comment]
From 2.4.0rc5 MA supports FLAC, AAC, MP3 and WAV. Not all players might be able to accept all codecs.

**Reply by marcelveldt** [reply]
The new ESPHome mediaplayer is also available since yesterday - MA will auto pickup the configuration for the esphome player.
So if you configure that to use FLAC 48khz, 16 bits, MA will use that audio format

**Comment by OzGav** [comment]
The feature request has been implemented successfully. As such this feature request will be closed and locked. If there are enhancements desired then open a new [feature request](https://github.com/orgs/music-assistant/discussions/categories/feature-requests-and-ideas). If there is a question about how this works then check the [documentation](https://beta.music-assistant.io/), or ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord. If there are problems then ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord or open an [issue](https://github.com/music-assistant/support/issues). Thanks for your interest in seeing this idea being brought to life.
